presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java

```diff
@@ -184,6 +184,9 @@ public OrcReader(OrcDataSource orcDataSource, OrcEncoding orcEncoding, DataSize
         try (InputStream footerInputStream = new OrcInputStream(orcDataSource.getId(), footerSlice.getInput(), decompressor, newSimpleAggregatedMemoryContext(), footerSize)) {
             this.footer = metadataReader.readFooter(hiveWriterVersion, footerInputStream);
         }
+        if (footer.getTypes().size() == 0) {
```

Comment:
Are such files illegal by ORC standard?

Reply:
The spec doesn't say explicitly say the file must contain types, but it is pretty clear that types are expected.  I suggest we make this change, and if someone comes with a real use case later we can change this code allow this.

Reply:
I agree with @dain . My understanding about spec is even when there is no column, the root of the type tree still exists (and is a struct of nothing).

https://orc.apache.org/specification/ORCv0/, section `Type Information`: 

> The type tree is flattened in to a list via a pre-order traversal where each type is assigned the next id. Clearly the root of the type tree is always type id 0. Compound types have a field named subtypes that contains the list of their children’s type ids.